include/libvfio-user.h

```diff
@@ -316,24 +313,28 @@ vfu_setup_device_reset_cb(vfu_ctx_t *vfu_ctx, vfu_reset_cb_t *reset);
  * Function that is called when the guest maps a DMA region. Optional.
  *
  * @vfu_ctx: the libvfio-user context
- * @iova: iova address
- * @len: length
+ * @iova: IOVA address and length
+ * @vaddr: virtual address and length if the region is mappable, which can be
+ *  different from IOVA due to alignment. If the region is not mappable then
+ *  @vaddr is set to NULL.
```

Comment:
nit, "@vaddr is NULL" ?

Reply:
should this mention that only the @iova iovec area should be accessed despite the mapping boundaries?

Reply:
> nit, "@vaddr is NULL" ?

Sure.

> should this mention that only the @iova iovec area should be accessed despite the mapping boundaries?

Definitely. It just occured to me that the user doesn't know where exactly the valid part of vaddr starts. E.g. when using 4K pages and the IOVA is 0x200-400, the vaddr will be e.g. 0xffff0000-0xffff1000. How will the user know that they should access vaddr at offset 0xffff0000+0x200? I think we need to also provide an offset argument in the callbacks?